build/Dockerfile

```diff
@@ -1,45 +1,38 @@
-
-RUN adduser --disabled-password --gecos '' fdb && chown -R fdb /opt && chmod -R 0777 /opt
-
-USER fdb
-
-# wget of bintray without forcing UTF-8 encoding results in 403 Forbidden
-RUN cd /opt/ && wget http://downloads.sourceforge.net/project/boost/boost/1.52.0/boost_1_52_0.tar.bz2 &&\
-    wget --local-encoding=UTF-8 https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2 &&\
-    echo '2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba boost_1_67_0.tar.bz2' | sha256sum -c - &&\
-    tar -xjf boost_1_52_0.tar.bz2 &&\
-    tar -xjf boost_1_67_0.tar.bz2 &&\
-    rm boost_1_52_0.tar.bz2 boost_1_67_0.tar.bz2
-
-USER root
-
-RUN pip install boto3==1.1.1
-
-RUN ln -s /usr/bin/nodejs /usr/bin/node
-
-RUN cd /opt/ && wget https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.6.4.tar.gz &&\
-    wget https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-2.6.4.tar.gz.asc &&\
-    wget https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl.asc &&\
-    gpg --import libressl.asc && gpg --verify libressl-2.6.4.tar.gz.asc libressl-2.6.4.tar.gz &&\
-    tar -xzf libressl-2.6.4.tar.gz && cd libressl-2.6.4 &&\
-    ./configure CFLAGS="-fPIC -O3" && make -j4 && make install &&\
-    cd /opt/ && rm -r libressl-2.6.4/ libressl-2.6.4.tar.gz libressl-2.6.4.tar.gz.asc libressl.asc
-
-RUN LANGUAGE=en_US.UTF-8 LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 locale-gen en_US.UTF-8
-
-RUN dpkg-reconfigure locales
-
-ARG TARGET_LIBC_VERSION=2.17
-ENV TARGET_LIBC_VERSION=$TARGET_LIBC_VERSION
-
-ARG CC=/usr/bin/gcc
-ENV CC=$CC
-
-ARG LIBRARY_PATH=/usr/local/lib
-ENV LIBRARY_PATH=$LD_FLAGS
+FROM centos:6
+LABEL version=0.0.4
+
+RUN yum install -y yum-utils
+RUN yum-config-manager --enable rhel-server-rhscl-7-rpms
+RUN yum -y install centos-release-scl
+RUN yum install -y devtoolset-7
+
+# install cmake
+RUN curl -L https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz > /tmp/cmake.tar.gz &&\
+    echo "563a39e0a7c7368f81bfa1c3aff8b590a0617cdfe51177ddc808f66cc0866c76  /tmp/cmake.tar.gz" > /tmp/cmake-sha.txt &&\
+    sha256sum -c /tmp/cmake-sha.txt &&\
+    cd /tmp && tar xf cmake.tar.gz && cp -r cmake-3.13.4-Linux-x86_64/* /usr/local/
+
+# install boost
+RUN curl -L https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2 > /tmp/boost.tar.bz2 &&\
+    cd /tmp && echo "2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba  boost.tar.bz2" > boost-sha.txt &&\
+    sha256sum -c boost-sha.txt && tar xf boost.tar.bz2 && cp -r boost_1_67_0/boost /usr/local/include/
+
+# install mono (for actorcompiler)
+RUN yum install -y epel-release
+RUN yum install -y mono-core
+
+# install Java
+RUN yum install -y java-1.8.0-openjdk-devel
+
+# install python
+RUN yum install -y rh-python36-python-devel
```

Comment:
Wouldn't this might be a good point in time to port all scripts to python3? Running python 2 and 3 in parallel is currently a huge pain - I would much rather only support one.

If you can point me to the scripts that only work on 2.7, I can try to port them.

Reply:
I just kind of assumed knowing that we have python scripts around, and haven't run with python3 yet.

Scraping files, I'm looking at:

```
bindings/python/tests/tester.py
bindings/python/tests/tuple_tests.py
packaging/osx/scripts-clients/postinstall
packaging/make_public.py
tests/python_tests/python_correctness.py
tests/python_tests/python_performance.py
build/vcxprojtom4.py
build/csprojtom4.py
layers/pubsub/ps_test.py
layers/pubsub/remotesubscribe.py
layers/pubsub/ps_prompt.py
layers/taskbucket/test.py
layers/pubsub/remotesend.py
layers/pubsub/remoteload.py
layers/pubsub/ps_tutorial.py
```

But they all just refer to `/usr/bin/python`, so no idea how many are python3 compatible already.  Of these,

```
packaging/make_public.py
build/vcxprojtom4.py
build/csprojtom4.py
```

Are the actually important ones, I think.

But, how about we handle that in a different PR?  I'm not suggesting long term support of both, but for now, I think it would be easier.

Reply:
That I agree with. The problem is I am currently not sure how to have two python versions cleanly in this docker image.

@preston4tw might be able to help here? Or maybe we don't even need python 3, I am not sure, but it could be that the TestRunner doesn't work in python 2.7 (at least it is untested)